addons/mail/static/src/core/message_model.js

```diff
@@ -192,7 +194,26 @@ export class Message {
         return htmlToTextContentInline(this.body);
     }
 
+    get notificationIcon() {
+        switch (this.notificationType) {
+            case "pin":
+                return "fa fa-thumb-tack";
+        }
+        return null;
+    }
+
     get failureNotifications() {
         return this.notifications.filter((notification) => notification.isFailure);
     }
+
+    get pinnedAt() {
+        if (!this.pinned_at) {
+            return null;
+        }
+        return luxon.DateTime.fromISO(new Date(this.pinned_at).toISOString());
+    }
+
+    get pinned() {
+        return Boolean(this.pinned_at);
+    }
```

Comment:
overkill?

Reply:
Agree, checking truthy `pinned_at` is enough